### ISSUE TITLE
mcrypt deprecated - Cronjob not working on PHP 7.1.7

### ISSUE THREAD

**Issue by mudolux** [phpservermon/phpservermon#491]
Hello,

Cronjob not working on PHP 7.1.7

```Fatal error: Uncaught Error: Call to undefined function mcrypt_get_iv_size() in /var/www/.../phpservermonitor/src/includes/functions.inc.php:717
Stack trace:
#0 /var/www/.../phpservermonitor/src/psm/Util/Server/Updater/StatusUpdater.php(221): psm_password_decrypt('...', '...')
#1 /var/www/.../phpservermonitor/src/psm/Util/Server/Updater/StatusUpdater.php(100): psm\Util\Server\Updater\StatusUpdater->updateWebsite(2)
#2 /var/www/.../phpservermonitor/src/psm/Util/Server/UpdateManager.php(71): psm\Util\Server\Updater\StatusUpdater->update('23')
#3 /var/www/.../phpservermonitor/cron/status.cron.php(77): psm\Util\Server\UpdateManager->run(true)
#4 {main}
  thrown in /var/www/.../phpservermonitor/src/includes/functions.inc.php on line 717```

**Comment by shyim** [comment]
I have also PHP 7.1.7 and it works. Did you forgot to install mcrypt extension?

**Comment by mudolux** [comment]
Hm yes, i haven't installed mcrypt_get_iv_size because the PHP Doc says its deprecated since 7.1.0.
See http://de2.php.net/manual/de/function.mcrypt-get-iv-size.php

**Comment by ryannix123** [comment]
I had to run sudo apt-get install mcrypt php7.1-mcrypt to get the cron jobs working again with PHP 7.1.9 on Ubuntu 16.0.4

**Comment by rAmfu** [comment]
Start with this your cronjob
1. crontab -e
2. Add this line: */5 * * * * /usr/bin/php7.0 /your/path/to/cron/status.cron.php

**/usr/bin/php7.0** not **/usr/bin/php**

**Comment by TimZ99** [comment]
mcrypt_get_iv_size is deprecated since 7.1.0. The goal is to always support the latest version of PHP, so we need to update the code.

**Comment by acrylic-style** [comment]
The best way to migrate from mcrypt is to use openssl.
Unfortunately, since it is incompatible with mcrypt, encryption needs to be restarted from the beginning, and "upgrade.php" may be necessary.
Or, although not recommended much, it is a method to install mcrypt using pecl.

**Comment by TimZ99** [comment]
There is an upgrade function so we could use that.

**Comment by rht0910** [comment]
https://github.com/rht0910/phpservermon/blob/develop/src/includes/functions.inc.php
How about this?

**Comment by jmcgranahan** [comment]
this is still an issue. I'm using PHP 7.4 and mcrypt_create_iv was removed in PHP 7.2 (https://www.php.net/manual/en/function.mcrypt-create-iv.php) so now I'm getting a butt load of errors and manually checking the status gives an error. One suggestion was to use random_bytes (https://www.php.net/manual/en/function.random-bytes.php)

**Comment by TimZ99** [comment]
Fixed in #805.
